### PR TITLE
docs: add mock API project examples

### PR DESCRIPTION
**Pull request by yury-s**
* The battery API example uses https://github.com/pazguille/demo-battery-api app.
* Filesystem API example uses its own test page.

#11670

### CODE REVIEW

**Review by dgozman — APPROVED** [review]

File: examples/mock-battery/tests/update-battery-status.spec.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// @ts-check
+const { test, expect } = require('@playwright/test');
+
+let log = [];
+
+test.beforeEach(async ({page}) => {
+  log = [];
+  await page.exposeFunction('logCall', msg => log.push(msg));
```
[the remark this answers is not in the document]
Comment (yury-s) [reply examples/mock-battery/tests/update-battery-status.spec.js:8]: > * wdyt about making this a fixture?

It's not a fixture in the docs which is why I keep it as `beforeEach` here.

> * some comments would be nice here.

Done.

[the remark this answers is not in the document]
Comment (yury-s) [reply examples/mock-battery/tests/update-battery-status.spec.js:8]: Also wrapping both logCall and logs in a fixture would require including `logs` into beforeEach parameters even though it's not actually used.

File: examples/mock-filesystem/tests/directory-reader.spec.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+// @ts-check
+const { test, expect } = require('@playwright/test');
+
+test.beforeEach(async ({page}) => {
+  await page.addInitScript(() => {
+    class FileSystemHandleMock {
+      constructor({name, children}) {
+        this.name = name;
+        children ??= [];
```
Comment (mxschmitt) [inline examples/mock-filesystem/tests/directory-reader.spec.js:9]: Do all browsers understand that already? Or does it go through babel?

Comment (yury-s) [reply examples/mock-filesystem/tests/directory-reader.spec.js:9]: dunno, tested with chrome only and the example doesn't run other browsers, we can change it to the old syntax

Comment (yury-s) [reply examples/mock-filesystem/tests/directory-reader.spec.js:9]: works in all browsers

**Pull request merged**